src/main/scala/rules/QuantifiedChunkSupport.scala

```diff
@@ -1589,6 +1603,19 @@ object quantifiedChunkSupporter extends QuantifiedChunkSupport {
       inverseFunctions(idx) = fun
       inversesOfFcts(idx) = inv(invertibles)
       inversesOfCodomains(idx) = inv(codomainQVars)
+
+      if (qvar.sort != sorts.Int && qvar.sort != sorts.Ref) {
+        // Types known to be infinite, thus there is no need to constrain the domain using image functions.
+        val imgFun = v.decider.fresh("img", (additionalInvArgs map (_.sort)) ++ invertibles.map(_.sort), sorts.Bool)
+        val img = (ts: Seq[Term]) => App(imgFun, additionalInvArgs ++ ts)
+
+        imageFunctions(idx) = imgFun
+        imagesOfFcts(idx) = img(invertibles)
+        imagesOfCodomains(idx) = img(codomainQVars)
+      } else {
+        imagesOfFcts(idx) = True()
```

Comment:
A comment `// imageFunctions(idx) remains null` couldn't hurt, and would help understanding the [filter in line 1693](https://github.com/viperproject/silicon/pull/666/files#diff-aed5f0e36948a8a0ce2bcd027936b8cbe7fbbb8fc68bf9ef2f2ea10ba8c82f2dR1693).